### PR TITLE
Add commentator privileges

### PR DESCRIPTION
**Pull request by Jeff-Thompson12**
Addresses #564 

Also adds a warning if any privileges are not assigned to any roles in the application.

### CODE REVIEW

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
@Jeff-Thompson12, nice work on this PR. I haven't run any code yet, but just from looking at the diff, I had a few suggestions below.

However, first I have a comment/question: I feel like the `overall_comment` privilege (access to overall comment and package summary UIs) should be added to the default privileges for the `reviewer` role. That way, a reviewer could be seen as the person doing most of the "grunt work". They are the person tasked with uploading a package, writing the report, and creating a short synopsis regarding a final decision. Then the "lead" can evaluate the review and make a final decision. What do you think?

Requesting changes for the reasons below:

(1) We should probably move away from using images to display the contents of the config file, since it is harder to update. Instead we should just use the triple back tick and `yml` keyword to generate these code blurbs.

image

(2) This vignette is weak sauce compared to the other privileges! 😆 

image

**Comment by Jeff-Thompson12** [comment]
> (2) This vignette is weak sauce compared to the other privileges! 😆

Is there a requested change here, or are you just mocking the low effort put in updating the vignette?

**Comment by AARON-CLARK** [comment]
> > (2) This vignette is weak sauce compared to the other privileges! 😆
> 
> Is there a requested change here, or are you just mocking the low effort put in updating the vignette?

Both? Could you please add an image or two to show off the UI elements impacted by these privileges? Or something along those lines. Perhaps even elaborate on why the two comment privileges are separate?

File: vignettes/User_Roles_and_Privileges.Rmd [hunk 1]
```diff
@@ -305,13 +305,21 @@ A user with the `delete_package` privilege can remove packages from consideratio
 
 ### Overall Commentator
 
-A user with the `overall_comment` privilege can add/edit the overall comment and summary of a package.
+A user with the `overall_comment` privilege can add/edit the overall comment and summary of a package. Unlike maintenance or community usage metrics, each package can only have one overall comment and summary. A user with the `overall_comment` privilege can provide the overall recommendation for the package decision and provide the summary of what the package offers.
+
```
Comment (AARON-CLARK) [inline vignettes/User_Roles_and_Privileges.Rmd:309]: Add a <br> here to create a little spacing between the paragraph and the image. But if you do that, you may want to do it preceding each image on this page/ vignette.

File: vignettes/images/start_decision_automation.png
Comment (AARON-CLARK) [inline vignettes/images/start_decision_automation.png:1]: You didn't like my arrow?

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
Hi @Jeff-Thompson12, looks good. I have a few comments/ suggestions below.

File: vignettes/images/start_decision_automation.png
Comment (Jeff-Thompson12) [reply vignettes/images/start_decision_automation.png:1]: Sorry. Was hoping to sneak that change in.

File: vignettes/User_Roles_and_Privileges.Rmd (hunk 1 above)
Comment (Jeff-Thompson12) [reply vignettes/User_Roles_and_Privileges.Rmd:330]: I added a <br> before and after images.

**Review by AARON-CLARK — APPROVED** [review]
👍🏼

**Pull request merged**